plugins/woocommerce-blocks/assets/js/base/components/select/index.tsx

```diff
@@ -22,61 +27,151 @@ type SelectProps = Omit<
 	options: SelectOption[];
 	label: string;
 	onChange: ( newVal: string ) => void;
+	errorId?: string;
+	required?: boolean;
+	errorMessage?: string;
 };
 
 export const Select = ( props: SelectProps ) => {
-	const { onChange, options, label, value, className, size, ...restOfProps } =
-		props;
-
+	const {
+		onChange,
+		options,
+		label,
+		value = '',
+		className,
+		size,
+		errorId,
+		required,
+		errorMessage = __( 'Please select a valid option', 'woocommerce' ),
+		placeholder,
+		...restOfProps
+	} = props;
 	const selectOnChange = useCallback(
 		( event: React.ChangeEvent< HTMLSelectElement > ) => {
 			onChange( event.target.value );
 		},
 		[ onChange ]
 	);
 
+	const emptyOption: SelectOption = useMemo(
+		() => ( {
+			value: '',
+			label:
+				placeholder ??
+				sprintf(
+					// translators: %s will be label of the country input. For example "country/region".
+					__( 'Select a %s', 'woocommerce' ),
+					label?.toLowerCase()
+				),
+			disabled: !! required,
+		} ),
+		[ label, placeholder, required ]
+	);
+	const errorIdString = errorId || restOfProps.id || '';
+	const optionsWithEmpty = useMemo< SelectOption[] >( () => {
+		return [ emptyOption ].concat( options );
+	}, [ options, emptyOption ] );
+
+	const { setValidationErrors, clearValidationError } =
```

Comment:
Overall i do think the validation being here is better. It'd be nice if we could opt out of the placeholder option but we don't actually have a need for that right now do we?

Reply:
Yes I can't think of a use case yet unless we start having default values but those should be set server-side.

Reply:
Added an option to opt out, if a field is required and has a value, we skip the empty one.

Reply:
> Added an option to opt out, if a field is required and has a value, we skip the empty one.

Is this possible to do from an extension's point of view?

Reply:
In what sense? Additional fields API? or someone using the select component?

They can simply pass a value other than `""`

Reply:
When passing empty option it still renders the placeholder and an empty option as well. I don't think this is a problem.

Reply:
Yeah this is up to the consumer, I think we can filter out empty options if we want. For additional fields API empty option is no longer passed.